### ISSUE TITLE
😔 Ferros mia? Webtorrent future?

### ISSUE THREAD

**Issue by Fenny** [webtorrent/webtorrent#1630]
Hi all,

I'm a long time webtorrent supporter on patreon [https://www.patreon.com/feross](https://www.patreon.com/feross) and really appreciate the work people put in this project.
However, the last year the activity has dropped enormous. Many open unanswered issue's and pull requests, I try to solve as many as possible.

@feross if you are reading this, maybe you could give the community and patreons an update on webtorrent and where it's going. We all know you are busy, but it would be nice if we know what the future holds regarding the webtorrent project. 🙌 

Thank you for your contribution to the community, not all heroes wear capes! 👙

**Comment by feross** [comment]
Thanks for the kind words, @Fenny.

I'll repost what I wrote in the [webtorrent-desktop](https://github.com/webtorrent/webtorrent-desktop/issues/1604#issuecomment-510160907) repo recently:

> I needed to step away from OSS for a while to rest and recover. I achieved a lot in a short time in the early days of WebTorrent by working really hard but I got totally burned out after several years of working on it full-time. (literally. I worked exclusively on WebTorrent and didn't even have a job to support me for many years). I now feel that I've recovered sufficiently to return with a healthier attitude. We'll see :)

Also, I appreciate the support from you and other patrons. I've released two new versions of `standard` and done lots of maintenance work on all the various packages that I maintain. While it's nice to get some funds for the work I've done, it's not really possible to live on this amount in the San Francisco Bay Area. I'd love to find a way to fund my work on open source full-time and get paid fairly. I have several ideas that I plan to try in the coming year – stay tuned.

Zooming out a bit, a big issue facing most OSS projects is that they rely on free work from contributors who have altruistic motivations. It seems to me that this can work for a while, but eventually folks will burn out, or stop contributing because there's only so much time for coding in a day they can't justify doing so much work for free. Some folks get a job and try to squeeze in time for OSS after work hours. But it's always a struggle. I wish that folks could get paid to work on OSS _instead_ of getting a job. What if working on OSS full-time was a viable alternative to getting a job? What if it was possible to make as much – or more – working on OSS? How awesome would our OSS projects become? We'd have so many more people contributing and participating.

I think that we as an OSS community need to find a way to get businesses to put way more money into the OSS that they rely on. If every business using WebTorrent were to pay $50/mo, we'd have enough funds to pay for tons of people to work on WebTorrent full-time. But there's no incentive or easy mechanism to do this at the moment. Anyway, this is a long discussion for another time...

To answer your question: I'll continue to do WebTorrent maintenance in my free time. And other team members are welcome to continue to do the same. I'm doing my best to catch up on issues and fix issues here and there when I have time. But until I figure out the funding situation, I probably won't be able to put in tons of time like I did in the early days.

One positive bit of news: I'm currently doing a paid contract for [Brave](https://brave.com) to improve their built-in WebTorrent integration, and I'm fixing some WebTorrent issues as part of that. I think that this is one good model for how to fund continued work on WebTorrent, and I think it's repeatable.
